lib/elixir/lib/calendar.ex

```diff
@@ -417,12 +489,273 @@ defmodule Date do
       2
       iex> Date.day_of_week(~N[2016-11-01 01:23:45])
       2
+
   """
-  @spec day_of_week(Calendar.date) :: non_neg_integer()
+  @spec day_of_week(Calendar.date | Calendar.integer_date) :: non_neg_integer()
   def day_of_week(%{calendar: calendar, year: year, month: month, day: day}) do
     calendar.day_of_week(year, month, day)
   end
 
+  def day_of_week(date) when is_integer(date) do
```

Comment:
> For now I'm going to update my code to use the new proposed intermediate format and resubmit and we can see where is goes from there - unless there is already a strong view?

That sounds a great idea but please don't "force push". Just add commits on top so we can revert the changes in the chance we find out that using `{RD/JD, fraction}` is a bad idea.